wavelength-plugins/exp.js

```diff
@@ -152,13 +140,27 @@ exports.commands = {
 	xp: 'exp',
 	exp: function (target, room, user) {
 		if (!this.runBroadcast()) return;
-		if (!target) target = user.name;
-
-		const targetId = toId(target);
-
-		EXP.readExp(targetId, exp => {
-			this.sendReplyBox('<b>' + WL.nameColor(targetId, true) + '</b> has ' + exp + ' exp and is level ' + WL.level(targetId) + ' and needs ' + WL.nextLevel(targetId) + ' to reach the next level.');
-		});
+		let targetId = toId(target);
+		if (target || !target && this.broadcasting) {
+			if (!target) targetId = user.userid;
+			EXP.readExp(targetId, exp => {
+				this.sendReplyBox('<b>' + WL.nameColor(targetId, true) + '</b> has ' + exp + ' exp and is level ' + WL.level(targetId) + ' and needs ' + WL.nextLevel(targetId) + ' to reach the next level.');
+			});
+		} else {
+			EXP.readExp(user.userid, exp => {
+				user.popup("|html|Name: " + WL.nameColor(user.userid, true) + "<br />Current level: " + WL.level(user.userid) + "<br />Exp Needed for Next level: " + WL.nextLevel(user.userid) +
+					"<br />All rewards have a 1 time use! <br /><br />" +
+					"Level 5 unlocks a free Custom Symbol. <br /><br />" +
+					"Level 10 unlocks a free Custom Avatar. <br /><br />" +
+					"Level 15 unlocks a free Profile Title. <br /><br />" +
+					"Level 20 unlocks a free Custom Userlist Icon. <br /><br />" +
+					"Level 25 unlocks a free Emote. <br /><br />" +
+					"Level 30 unlocks a free Custom Color.  <br /><br />" +
+					"Level 35 unlocks 50 " + currencyPlural + ". <br /><br />" +
+					"Level 40 unlocks a free chatroom. <br /><br />"
+				);
+			});
```

Comment:
This would be better if it was a reply box rather than a popup.

Reply:
well this is for the user only i plan on modifying it later to show what they have unlocked and what they have not unlocked but this is just the basis of that

Reply:
but then again it makes no difference if they cant broadcast the bigger reply box

Reply:
done